src/main/java/com/google/cloud/partners/pubsub/kafka/SubscriberImpl.java

```diff
@@ -135,33 +136,34 @@ public void createSubscription(
       Subscription request, StreamObserver<Subscription> responseObserver) {
     try {
       validateCreation(request);
-      SubscriptionProperties subscriptionProperties = buildSubscriptionProperty(request);
-      Configuration.getApplicationProperties()
-          .getKafkaProperties()
-          .getConsumerProperties()
-          .getSubscriptions()
-          .add(subscriptionProperties);
-
-      SubscriptionManager subscriptionManager =
-          subscriptionManagerFactory.create(
-              subscriptionProperties, kafkaClientFactory, commitExecutorService);
-
-      subscriptions.put(subscriptionProperties.getName(), subscriptionManager);
-      statisticsManager.addSubscriberInformation(subscriptionProperties);
-
+      createSubscription(buildSubscriptionProperty(request));
       responseObserver.onNext(request);
       responseObserver.onCompleted();
     } catch (StatusException e) {
       responseObserver.onError(e);
     }
   }
 
+  public void createSubscription(SubscriptionProperties subscriptionProperties) {
```

Comment:
I'm happy with this change but let's rename it to createSubscriptionFromProperties to distinguish it from the public Pub/Sub API method. It probably wouldn't hurt to make it package-private as well.

Reply:
How's `addSubscription`? :)